src/Database/PG/Query/Pool.hs

```diff
@@ -49,11 +50,25 @@ import qualified Data.HashTable.IO             as HI
 import qualified Data.Pool                     as RP
 import qualified Data.Text                     as T
 import qualified Database.PostgreSQL.LibPQ     as PQ
-
-type PGPool = RP.Pool PGConn
+import System.Metrics.Distribution (Distribution)
+import qualified System.Metrics.Distribution as EKG.Distribution
+
+data PGPool = PGPool
+  { -- | the underlying connection pool
+    _pool :: RP.Pool PGConn,
+    -- | EKG stats about how we acquire, release, and manage connections
+    _stats :: PGPoolStats
+  }
+
+-- | Actual ekg gauges and other metrics are not created here, since those depend on
+-- a store and it's much simpler to perform the sampling of the distribution from within graphql-engine.
+newtype PGPoolStats = PGPoolStats
+  { -- | time taken to acquire new connections from postgres
+    _connAcquireTime :: Distribution
```

Comment:
If I understand #798 correctly, we want the time taken to acquire connections from the connection pool. Is that right?

Although, I think its a great idea to capture the time to postgres as well. We should definitely have this. But, I think we have to add another `Distribution` to track the same thing but for connections from pool.

Possibly in this function: https://github.com/hasura/pg-client-hs/blob/swann/pool-metrics/src/Database/PG/Query/Pool.hs#L273 ?